// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package related_website_sets.proto;

message AllInputs {
  required PublicSets public_sets = 1;
  required Policy policy = 2;
  required CommandLineSwitch command_line_switch = 3;
}

message PublicSets {
  repeated Set sets = 1;
}

message Policy {
  repeated Set replacements = 1;
  repeated Set additions = 2;
}

message CommandLineSwitch {
  optional Set set = 1;
}

message Set {
  required Site primary = 1;
  repeated Site associated = 2;
  repeated Site service = 3;
  repeated SitePair cctld_aliases = 4;
}

message SitePair {
  required Site alias = 1;
  required Site canonical = 2;
}

message Site {
  optional SubdomainIndex subdomain_index = 1;
  required SiteIndex site_index = 2;
  required Tld tld = 3;
}

enum SubdomainIndex {
  SUBDOMAIN_0 = 0;
  SUBDOMAIN_1 = 1;
  SUBDOMAIN_2 = 2;
}

enum SiteIndex {
  SITE_0 = 0;
  SITE_1 = 1;
  SITE_2 = 2;
  SITE_3 = 3;
  SITE_4 = 4;
}

enum Tld {
  TLD_0 = 0;
  TLD_1 = 1;
}
